### PR TITLE
Support for blocking requests to specific domains

### PR DESCRIPTION
**Pull request by ivanvanderbyl**
This PR adds support for blocking requests to specific domains. This can be used to prevent sending traffic to analytics providers or creating load against services which are outside of our concern.

## Usage

```ts
export const settings: TestSettings = {
  blockedDomains: ['*.google-analytics.com', 'bootstrapcdn.com', '*:80'],
}
```

**Debugging:**

```shell
DEBUG=network:intercept element run script.ts
```

### CODE REVIEW

File: packages/element/src/network/Interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	async detach(page: Page) {
+		page.off('request', this.requestBlocker)
+		await page.setRequestInterception(false)
+	}
+
+	private requestBlocker = (interceptedRequest: Request) => {
+		let url = new URL(interceptedRequest.url())
+
+		if (this.blockedDomains.some(domain => url.hostname.includes(domain))) {
```
Comment (90kts) [inline packages/element/src/network/Interceptor.ts]: wonder if `match` would work better than `includes`, as I think people might like to specify wildcards e.g. `*.cdn.com.au`

Comment (lachie) [reply packages/element/src/network/Interceptor.ts]: was my thought that this'd be the first thing users'd expect

Comment (ivanvanderbyl) [reply packages/element/src/network/Interceptor.ts]: Yeah good point. I guess we could make it a RegExp or string

Comment (ivanvanderbyl) [reply packages/element/src/network/Interceptor.ts]: `string.match(...)` won't really deliver what we expect here if we want to allow usual assortment of wildcards like `*.foo.com` as it's not a valid regex. I'll try the Node package https://www.npmjs.com/package/url-pattern#make-pattern-from-string

Comment (ivanvanderbyl) [reply packages/element/src/network/Interceptor.ts]: Actually micomatch seems like a better solution https://github.com/micromatch/micromatch

Comment (ivanvanderbyl) [reply packages/element/src/network/Interceptor.ts]: Done

**Pull request merged**